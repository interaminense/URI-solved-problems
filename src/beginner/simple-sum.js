/**
 * @author Adriano Interaminense <adriano.interaminense@gmail.com>
 *
 * Read two integer values, in this case, the variables A and B. After this,
 * calculate the sum between them and assign it to the variable SOMA. Write the
 * value of this variable.
 *
 * INPUT
 *
 * The input file contains 2 integer numbers.
 * @param {number} A
 * @param {number} B
 *
 * OUTPUT
 *
 * Print the variable SUM with all the capital letters, with a blank space before
 * and after the equal signal followed by the corresponding value to the sum of A and B.
 */

const tmpl = result => `SUM = ${result}`;

const sum = (A, B) => tmpl(A + B);

export { sum, tmpl };
