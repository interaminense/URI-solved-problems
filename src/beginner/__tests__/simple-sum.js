import { sum, tmpl } from '../simple-sum';

describe('Simple sum', () => {
  let A = 0;
  let B = 0;

  it('The sum 30 + 10 must be 40', () => {
    A = 30;
    B = 10;

    expect(sum(A, B)).toEqual(tmpl(40));
  });

  it('The sum -30 + 10 must be -20', () => {
    A = -30;
    B = 10;

    expect(sum(A, B)).toEqual(tmpl(-20));
  });

  it('The sum 0 + 0 must be 0', () => {
    A = 0;
    B = 0;

    expect(sum(A, B)).toEqual(tmpl(0));
  });
});
