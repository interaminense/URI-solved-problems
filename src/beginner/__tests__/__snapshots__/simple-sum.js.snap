// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple sum The sum -30 + 10 must be -20 1`] = `"SUM = -20"`;

exports[`Simple sum The sum 0 + 0 must be 0 1`] = `"SUM = 0"`;

exports[`Simple sum The sum 30 + 10 must be 40 1`] = `"SUM = 40"`;
